### ISSUE TITLE
Create a new endpoint on the backend to return a specific show

### ISSUE THREAD

**Issue by DTFsquad** [DTFsquad/Kodiri-Kodflix-MERN#32]
At the moment, we have a single endpoint (/rest/shows), returning the list of all shows.

This is fine for the landing page (localhost:3000), but not really for the Details component, as we have to:

1. [Backend] Return the whole list of TV shows (we may have thousands, so we may ruin the customer data plan!)
2. [Frontend] Filter them out, to just track the show matching the id available on the URL.
You can smell how inefficient this is. Let's start talking about filtering, so, on the Details component, we:

1. [Backend] Return the single TV show from (the one matching the id available on the URL).
2. [Frontend] Stop filtering, as we've done that already on the backend side.

**Technical guidance:**

1. Create a new dynamic route on the backend, alongside the /rest/shows. Have a look to the findOne Mongo method to return a single return, based on the provided query.

2. On the UI, just stop filtering the list of shows (we'll receive just one from now on, so there will be nothing to filter any more!).

Before the changes, the /rest/shows took, in my case, 1.7KB of data:
![image](https://user-images.githubusercontent.com/36204941/66917078-987f4300-f014-11e9-8c60-1ea84bd30b8e.png)
After the changes, I just had to transfer 451B of data
![image](https://user-images.githubusercontent.com/36204941/66917105-a59c3200-f014-11e9-99c6-4442c85cf865.png)
In other words, now we get the data ~4x faster, and we are using less bandwidth from the user data plan!
